packages/flutter/lib/src/painting/text_span.dart

```diff
@@ -345,46 +397,39 @@ class TextSpan extends DiagnosticableTree {
         && typedOther.style == style
         && typedOther.recognizer == recognizer
         && typedOther.semanticsLabel == semanticsLabel
-        && listEquals<TextSpan>(typedOther.children, children);
+        && listEquals<InlineSpan>(typedOther.children, children);
   }
 
   @override
-  int get hashCode => hashValues(style, text, recognizer, semanticsLabel, hashList(children));
+  int get hashCode => hashValues(text, style, recognizer, semanticsLabel, hashList(children));
```

Comment:
generally for operator== and hashCode we'd have the superclass responsible for its bits, and have the subclass only add more.